packages/server/lib/automation/automation.ts

```diff
@@ -22,7 +22,7 @@ export class Automation {
   private cookies: Cookies
   private screenshot: { capture: (data: any, automate: any) => any }
 
-  constructor (cyNamespace: string, cookieNamespace: string, screenshotsFolder: string, public onBrowserPreRequest: OnBrowserPreRequest) {
+  constructor (cyNamespace?: string, cookieNamespace?: string, screenshotsFolder?: string | false, public onBrowserPreRequest?: OnBrowserPreRequest) {
```

Comment:
Typing `server` is going to be a long and difficult journey. While these are not actually null in practice, according to our types they *can* be. We will need to find some way type this more strongly.

Reply:
At quick glance, it looks like everywhere we are creating the Automation class (other than tests 🤦 ), we're just shuttling along members off the `cfg` object. Maybe simpler to change to `new Automation(cfg)`?
<img width="425" alt="Screen Shot 2021-07-08 at 7 09 01 AM" src="https://user-images.githubusercontent.com/154748/124912009-6cabaa00-dfbb-11eb-96ed-cd5fd986f0fc.png">

Reply:
I feel like this is a slippery slope - just passing a huge config object around doesn't seem like a great idea. Unless there is a compelling reason to do so, I think it makes more sense to pass what we need (and nothing else). Eventually, we can probably refactor this class more so that not everything is nullable and more strongly typed.